### ISSUE TITLE
List of entities not updated in a relationship property

### ISSUE THREAD

**Issue by salvalacruz** [huridocs/uwazidocs#6244]
**Describe the bug**
When create entities, the list of entities to select in relationship properties are not updated in two cases:
- When adding an entity from a submit form and after, create an entity that uses this data in a relationship property.
- When deleting an entity and after, create an entity that uses this data in a relationship property.

**To Reproduce**
Case 1 - Adding data with a submit form:
1. Go to a submit form and add an entity, as usual.
2. Click on "Create entity" and select a template with a relationship propperty related with the templates from the first form.
3. See error: the added entity with the form does not appear in the list of entities.
4. Once you refresh the browser screen, the entity appears correctly.

Case 2 - Deleting an entity:
1. Delete any entity from a template related with a relationship property in another template.
2. Click on "Create entity" and select a template with a relationship propperty related with the deleted entity.
3. See error: the deleted entity still appears in the list of entities.
4. If you select this entity and save, Uwazi gets locked.
5. Once you refresh the browser screen, the deletd entity does not appear.

**Expected behavior**
When creating an entity, if it has relationship properties, the list of entities to be selected must be updated.

**Screenshots**
Case 1:
![image](https://github.com/huridocs/uwazi/assets/73410334/3ab8624b-edda-442a-a5c7-c76f84190ccf)

Case 2:
![image](https://github.com/huridocs/uwazi/assets/73410334/addcc3f5-bf65-4022-85bb-277e55aecd33)

**Device (please select all that apply)**
 - [X] Desktop
 - [x] Mobile

**Browser**
Tested with Brave and Chrome.

**Additional context**
This bug is not easy to explain, please let me know if it's better to meet and show it with a demo.

**Comment by RafaPolit** [comment]
This is, indeed, a known bug, but it's a tricky one.  Because the service that populates the sidepanel aggregations is not instant.  It can take a few seconds or a few minutes to get the newly created or deleted entity properly indexed in Elastic Search.  So we could either poll the remote server and try to update the value, or leave it as is and display a message saying that the side panel is not necessarily fully updated.

At any rate, the info is there, if you wait a couple of seconds and refresh, you will see that both Person 4 appears and Person 1 disappears.

So, this is a known bug, we could mitigate informing the user that the data may be off by a small margin, or we could leave it as is.  Making it "automatically appear" takes more effort than what we could put right now into the issue, I'm afraid.

**Comment by salvalacruz** [comment]
Thank you, anyway please notice that once adding a submit form or deleting an entity, if you refresh the browser and go straight to Create Entity, the list of entities is properly updated, with no seconds or minutes waiting.

**Comment by RafaPolit** [comment]
Yeah, the "seconds" or "minutes" could be milliseconds :P

**Comment by salvalacruz** [comment]
But the "Create Entity screen" does NOT refresh de list of entities in relationship properties until you refresh de browser. I think it does not depend on if you take secons or hours to do it.
I mean, the problem is that always is needed to refresh the browser.
Please let me know if you would like to meet and I show it to you, thanks!

**Comment by RafaPolit** [comment]
Values for Selects and Multiselects NEED to be reloaded upon entering Entity creation or edition.

**Comment by salvalacruz** [comment]
> Values for Selects and Multiselects NEED to be reloaded upon entering Entity creation or edition.

Is it posible if Relationship properties are reloadoed too, upon entering Entity creation or edition?

**Comment by RafaPolit** [comment]
This requires refreshing of the available lists and also backend validation that won't accept an entity that is no longer there.

**Comment by fnocetti** [comment]
> and also backend validation that won't accept an entity that is no longer there.
I closed the issue opened for the backend validation since it's not reproducible and upon checking, the code looks like it does what it needs to do.

Adding to the issue description, from my tests it looks like the client is not gracefully handling the 422 error returned by the backend, which is leading to the "Save" button getting locked.
I also tried with a Public form, and it shows an error without much detail. If you notice and fix the problem (which is unlikely to happen since the missing entity still appears in the list) you'll be able to continue.
